File: CSharp/Tests/MissingNumber.cs
using Solutions;

namespace Tests;

public class MissingNumberTest {
    [Fact]
    public void MissingNumber_Case1() {
        int[] nums = [3, 0, 1];
        int want = 2;

        int result = MissingNumber.MissingNumber1(nums);
        Assert.Equal(want, result);
    }

    [Fact]
    public void MissingNumber_Case2() {
        int[] nums = [0, 1];
        int want = 2;

        int result = MissingNumber.MissingNumber1(nums);
        Assert.Equal(want, result);
    }

    [Fact]
    public void MissingNumber_Case3() {
        int[] nums = [9, 6, 4, 2, 3, 5, 7, 0, 1];
        int want = 8;

        int result = MissingNumber.MissingNumber1(nums);
        Assert.Equal(want, result);
    }
}

File: Rust/tests/missing_number.rs
#[cfg(test)]
mod tests {
    use leetcode::missing_number::*;

    #[test]
    fn test_missing_number_case1() {
        let nums = vec![3, 0, 1];
        let want = 2;

        let result = missing_number(nums);

        assert_eq!(result, want);
    }

    #[test]
    fn test_missing_number_case2() {
        let nums = vec![0, 1];
        let want = 2;

        let result = missing_number(nums);

        assert_eq!(result, want);
    }

    #[test]
    fn test_missing_number_case3() {
        let nums = vec![9, 6, 4, 2, 3, 5, 7, 0, 1];
        let want = 8;

        let result = missing_number(nums);

        assert_eq!(result, want);
    }
}
